doc/flow_counters/routes_flow_counters.md

```diff
@@ -0,0 +1,475 @@
+"1.1.1.0/24:Vrf_default":"1.1.0.0/16:Vrf_default"
+"1.1.7.7/32:Vrf_default":"1.1.0.0/16:Vrf_default"
+}
+```
+
+### SAI API
+
+No new SAI API is required for this feature.
+
+SAI APIs shall be used in this feature:
+
+ |API                 |Function                           |
+ |:------------------:|-----------------------------------|
+ |sai_counter_api_t   |create_counter                     |
+ |                    |remove_counter                     |
+ |                    |get_counter_stats_ext              |
+ |sai_route_api_t     |set_route_entries_attribute        |
+
+SAI attributes shall be used in this feature:
+
+ |Struct                |Attribute                          |
+ |:--------------------:|-----------------------------------|
+ |sai_counter_stat_t    | SAI_COUNTER_STAT_PACKETS          |
+ |                      | SAI_COUNTER_STAT_BYTES            |
+ |sai_route_entry_attr_t| SAI_ROUTE_ENTRY_ATTR_COUNTER_ID   |
+
+### Configuration and management
+
+#### CLI/YANG model Enhancements
+
+##### CLI
+
+Enable/disable configuration:
+
+```
+counterpoll flowcnt-route  <enable | disable>
+Example:
+admin@sonic:~$ counterpoll flowcnt-route enable
+```
+
+Polling interval configuration:
+
+```
+counterpoll flowcnt-route interval <time_in_msec> // default - 1000ms
+Example:
+admin@sonic:~$ counterpoll flowcnt-route interval 2000
+```
+
+Show configuration:
+
+```
+counterpoll show
+Example:
+admin@sonic:~$ counterpoll show
+Type                              Interval (in ms)          Status
+--------------------------        ------------------        --------
+FLOW_CNT_ROUTE_STAT               default(1000)             disable
+```
+
+Config route pattern:
+
+```
+config flowcnt-route pattern <add | remove> [--vrf <vrf>] [--max <route-max>] <prefix-pattern>    // configure route pattern
+Example:
+admin@sonic:~$ config flowcnt-route pattern add --vrf Vrf_management --max 50 2.2.0.0/16
+Route Pattern Flow Counter configuration is successful
+
+admin@sonic:~$ config flowcnt-route pattern remove --vrf Vrf_management 2.2.0.0/16
+Route Pattern Flow Counter configuration is successful
+```
+
+Show configuration:
+
+```
+show flowcnt-route config
+Example:
+admin@sonic:~$ show flowcnt-route config
+Route pattern          VRF                 Max
+-----------------------------------------------
+3.3.0.0/16             Vrf_default         50
+```
+
+Show counters value:
+
+```
+show flowcnt-route stats   // show statistics of all route flow counters
+Example:
+admin@sonic:~$ show flowcnt-route stats
+Route pattern       VRF               Matched routes           Packets          Bytes
+--------------------------------------------------------------------------------------
+3.3.0.0/16          Vrf_default       3.3.1.0/24               100              4543
+                                      3.3.2.3/32               3443             929229
+                                      3.3.0.0/16               0                0
+
```

Comment:
How does the route pattern match work against the default route?

Reply:
0.0.0.0 and :: will be treat as exact match instead of pattern match. I will update this.